packages/shared/src/hooks/useFeedLayout.ts

```diff
@@ -1,45 +1,66 @@
-import { useMemo } from 'react';
-import { SharedFeedPage } from '../components/utilities';
-import { AllFeedPages } from '../lib/query';
-import { useActiveFeedNameContext } from '../contexts/ActiveFeedNameContext';
 import { useViewSize, ViewSize } from './useViewSize';
+import { useActiveFeedNameContext } from '../contexts/ActiveFeedNameContext';
+import { FeedPage, FeedPageLayoutMobile } from '../components/utilities';
+import { AllFeedPages } from '../lib/query';
+
+interface UseFeedLayoutReturn {
+  shouldUseMobileFeedLayout: boolean;
+  FeedPageLayoutComponent: React.ComponentType;
+}
 
 interface UseFeedLayoutProps {
-  feedName?: AllFeedPages;
   feedRelated?: boolean;
 }
 
-interface UseFeedLayout {
-  shouldUseMobileFeedLayout: boolean;
+export enum FeedPagesWithMobileLayout {
+  MyFeed = 'my-feed',
+  Popular = 'popular',
+  Search = 'search',
+  Upvoted = 'upvoted',
+  Discussed = 'discussed',
+  Bookmarks = 'bookmarks',
+  UserPosts = 'user-posts',
+  UserUpvoted = 'user-upvoted',
+  SquadPage = 'squads[handle]',
+  TagPage = 'tags[tag]',
+  SourcePage = 'sources[source]',
+}
```

Comment:
I'll repeat my concern from previous PR - we already have these constants defined elsewhere, e.g. in SharedFeedPage and OtherFeedPage enums. Can we reuse them here instead of redefining?

Reply:
See this [comment](https://github.com/dailydotdev/apps/pull/2990#discussion_r1565441374), let me know what u think 🙏

Reply:
Technically, do we even need a separate enum type? I see it only used in `checkShouldUseMobileFeedLayout` and then in `feedNameToHeading`. Could we rely on `AllFeedPages | UserProfileFeed` type instead? 

Also, consider this non-blocking :)

Reply:
Something like this could replace my enum:
export const SharedFeedPageArray = new Set(
SharedFeedPage
);
export const OtherFeedPageArray = new Set(
  Object.values(OtherFeedPage).filter(
    (feedPage) => feedPage !== OtherFeedPage.Author && feedPage !== OtherFeedPage.Notifications........,
  ),
);

FeedLayoutMobileFeedPages = [...SharedFeedPageArray, ...OtherFeedPageArray]

Reply:
You could re-use the enum, but pick what you need right? (atleast there is no confusion about what enum is the main one then)
I think this last nuance is the issue here.

Reply:
Think I can find another way, give me a sec

Reply:
Yeah, no whatever I think of I still end up with the above filtering.

If u think that is best I can change, else I think we should be good with the new enum

Reply:
I think it's mainly about not adding a new enum, but rather a filter that uses the existing one.
It will make it clear there's only one source (enum) of feed pages

Reply:
Have changed this to reuse the enums :D